client/components/domain-diagnostic/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+export default React.createClass( {
+
+	displayName: 'DomainDiagnostic',
+
+	propTypes: {
+		domain: React.PropTypes.string.isRequired,
+	},
+	getInitialState: function() {
+		return {
+			domainDidFail: false
+		};
+	},
+
+	componentDidMount() {
+		const self = this;
+		const imageChecker = new Image();
+		imageChecker.onerror = function() {
+			self.setState( { domainDidFail: true } );
+		}
+		imageChecker.src = 'http://' + this.props.domain + '/w32.gif';
```

Comment:
> If we do this server side we can do more comprehensive checks and also consider

Of course we could and I agree!
What do you imagine timeline for this could be?